asset/src/filter/processor.ts

```diff
@@ -0,0 +1,188 @@
+
+        this.functionName = toCamelCase(this.opConfig.filter_by);
+        this.filterValue = this._filterValueToArray();
+
+        // convert regexps on start up
+        if (this.opConfig.exception_rules) {
+            this.exception_rules = this._updateRegexRules();
+        }
+    }
+
+    filter(doc: DataEntity) {
+        if (this.exception_rules && this._exceptionCheck(doc)) return true;
+
+        const result = this._criteriaCheck(this._getDocValue(doc));
+
+        const keep = this.opConfig.invert === true ? result : !result;
+
+        if (this.opConfig.drop_to_dlq && keep === false) {
+            const err = new Error('Record was rejected');
+            this.rejectRecord(doc, err);
+        }
+
+        return keep;
+    }
+
+    _criteriaCheck(docValue: any | any[]) {
+        if (Array.isArray(docValue)) {
+            return docValue.some((value) => this._compareValues(value));
+        }
+
+        return this._compareValues(docValue);
+    }
+
+    _getDocValue(doc: DataEntity) {
+        if (Array.isArray(this.opConfig.field)) {
+            return this.opConfig.field.map((f) => get(doc, f));
+        }
+
+        if (this.functionName === 'size') {
+            if (this.opConfig.field === 'doc') {
+                return this._getSize(doc);
+            }
+
+            return this._getSize(doc[this.opConfig.field]);
+        }
+
+        if (this.opConfig.array_index > -1) {
+            if (Array.isArray(doc[this.opConfig.field])) {
+                return doc[this.opConfig.field][this.opConfig.array_index];
+            }
+
+            return undefined;
+        }
+
+        return get(doc, this.opConfig.field);
+    }
+
+    _filterValueToArray() {
+        if (Array.isArray(this.opConfig.value)) {
+            return this.opConfig.value;
+        }
+
+        return [this.opConfig.value];
+    }
+
+    _compareValues(docValue: any) {
+        return this.filterValue.some(
+            // @ts-expect-error
+            (filter) => this[this.functionName](docValue, filter)
+        );
+    }
+
+    match(docValue: any, filterValue: any) {
+        return docValue === filterValue;
+    }
+
+    regex(docValue: any, filterValue: any) {
+        const reg = new RegExp(filterValue, this.opConfig.regex_flags);
+        return reg.test(docValue);
+    }
+
+    ipRange(docValue: string, filterValue: string) {
+        return FieldValidator.inIPRange(docValue, {}, { cidr: filterValue });
+    }
+
+    validator(docValue: any) {
+        const fn = get(FieldValidator, this.opConfig.data_mate_function as string);
+        return fn(docValue, {}, this.opConfig.data_mate_args);
+    }
+
+    size(sizeValue: number) {
+        return sizeValue > this.opConfig.value;
+    }
+
+    _getSize(doc: any) {
+        // TODO: what about symbol and bigint ??
+        const typeSizes = {
+            undefined: () => 0,
+            boolean: () => 4,
+            number: () => 8,
+            bigint: () => 8,
```

Comment:
a symbol does not have a predefined byte size, and its length is undefined, granted its highly suspect that this can be in a record that we are processing